python/cudf/cudf/tests/test_udf_masked_ops.py

```diff
@@ -682,6 +690,391 @@ def f(x):
+def test_string_udf_rfind(data, substr):
+    # tests the `find` method of strings
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.rfind(substr)
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@string_udf_test
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "AI",
+                "gpu",
+                "2022",
+                "cuDF",
+                "again_gpu",
+            ]
+        }
+    ],
+)
+@pytest.mark.parametrize("substr", ["a", "cu", "", "12"])
+def test_string_udf_contains(data, substr):
+    # tests the boolean operation `substr in str`
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return substr in st
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@string_udf_test
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "AI",
+                "gpu",
+                "2022",
+                "cuDF",
+                "again_gpu",
+            ]
+        }
+    ],
+)
+@pytest.mark.parametrize("other", ["cudf", "123", "", " "])
+@pytest.mark.parametrize("cmpop", comparison_ops)
+def test_string_udf_cmpops(data, other, cmpop):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return cmpop(st, other)
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@string_udf_test
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "AI",
+                "gpu",
+                "2022",
+                "cuDF",
+                "again_gpu",
+            ]
+        }
+    ],
+)
+def test_string_udf_isalnum(data):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.isalnum()
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@string_udf_test
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "AI",
+                "",
+                "  ",
+                "12 ab",
+                "@2a",
+            ]
+        }
+    ],
+)
+def test_string_udf_isalpha(data):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.isalpha()
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@string_udf_test
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "123",
+                "",
+                "  ",
+                "12 ab",
+                "@2a",
+            ]
+        }
+    ],
+)
+def test_string_udf_isdigit(data):
+    data = cudf.DataFrame(data)
```

Comment:
We keep re-using the same kinds of test data with minor variations in the characters, capitalization, and whitespace. Should we make one dataframe with a superset of all the string inputs and define it as a fixture to be used by all tests here, rather than writing a `parametrize` that only has one input dataframe? We only care that pandas and cudf give the same results via `run_masked_udf_test`.